src/main/java/Parser.java

```diff
@@ -0,0 +1,199 @@
+/**
+ *  Parses user input.
+ */
+public class Parser {
+    /**
+     * Parse method to handle all types of commands.
+     * @param inputLine Description of the input command.
+     * @return Command object that has been specified by input file.
+     * @throws DukeException if there is an issue.
+     */
+    public static Command parse(String inputLine) throws DukeException {
```

Comment:
This method is a little long. Perhaps you could consider making smaller methods such as handleDeadline() or handleTodo() for example to make the code cleaner and more organised.